yt_dlp/extractor/redgifs.py

```diff
@@ -80,15 +75,196 @@ def _real_extract(self, url):
+        else:
+            entries = OnDemandPagedList(
+                functools.partial(self._fetch_page, query_str, api_query),
+                self._PAGE_SIZE)
+        title = tags
+        description = f'RedGifs search for {tags}, ordered by {order}'
+
+        return self.playlist_result(
+            entries,
+            playlist_id=query_str,
+            playlist_title=title,
+            playlist_description=description
+        )
+
+
+class RedGifsUserIE(RedGifsIE):
+    IE_DESC = 'Redgifs user'
+    _VALID_URL = r'https?://(?:www\.)?redgifs\.com/users/(?P<username>[^/?#]+)(?:\?(?P<query>.*))?'
+    _PAGE_SIZE = 30
+    _TESTS = [
+        {
+            'url': 'https://www.redgifs.com/users/lamsinka89',
+            'info_dict': {
+                'id': 'lamsinka89',
+                'title': 'lamsinka89',
+                'description': 'RedGifs user lamsinka89'
+            },
+            'playlist_mincount': 100,
+        },
+        {
+            'url': 'https://www.redgifs.com/users/lamsinka89?page=3',
+            'info_dict': {
+                'id': 'lamsinka89?page=3',
+                'title': 'lamsinka89',
+                'description': 'RedGifs user lamsinka89'
+            },
+            'playlist_count': 30,
+        },
+        {
+            'url': 'https://www.redgifs.com/users/lamsinka89?order=best&type=g',
+            'info_dict': {
+                'id': 'lamsinka89?order=best&type=g',
+                'title': 'lamsinka89',
+                'description': 'RedGifs user lamsinka89'
+            },
+            'playlist_mincount': 100,
+        }
+    ]
+
+    def _fetch_page(self, video_id, username, api_query, page=1):
+        api_query['page'] = page
+        data = self._download_json(
+            f'https://api.redgifs.com/v2/users/{username}/search',
+            video_id,
+            query=api_query
+        )
+        if 'error' in data:
+            raise ExtractorError(f'RedGifs said: {data["error"]}', expected=True)
+
+        return [self._parse_gif_data(entry) for entry in data['gifs']]
+
+    def _real_extract(self, url):
+        match = self._match_valid_url(url)
+        username = match.group('username')
+        query_str = match.group('query')
+        playlist_id = username
+        if query_str:
+            playlist_id = f'{username}?{query_str}'
+        description = f'RedGifs user {username}'
+
+        if not username:
+            raise ExtractorError('Invalid username', expected=True)
+
+        api_query = {
+            'order': 'recent',
+        }
+        page = None
+        if query_str:
+            query = compat_parse_qs(query_str)
+            if query.get('order'):
+                api_query['order'] = query.get('order', ('recent',))[0]
+                description += ", ordered by {order}"
+            if query.get('type'):
+                api_query['type'] = query.get('type')[0]
+            page = query.get('page', (1,))[0]
+
```

Comment:
lots of redundant code
```suggestion
        query = compat_parse_qs(query_str or '')
        api_query = {
            'order': query.get('order', ('recent',))[0],
        }
        if query.get('type'):
            api_query['type'] = query.get('type')[0]
        page = query.get('page', (None,))[0]
        
        description = f'RedGifs user {username}', ordered by {api_query["order"]}'
```

Reply:
Ah yes, that's a lot prettier. Changed that, thanks